custom_components/zwave_mqtt/__init__.py

```diff
@@ -63,60 +65,70 @@ async def mark_platform_loaded(platform):
 
     @callback
     def send_message(topic, payload):
-        _LOGGER.debug("sending message to topic %s", topic)
         mqtt.async_publish(hass, topic, json.dumps(payload))
 
     options = OZWOptions(send_message=send_message, topic_prefix=f"{TOPIC_OPENZWAVE}/")
     manager = OZWManager(options)
 
     for component in PLATFORMS:
-        hass.async_create_task(
-            hass.config_entries.async_forward_entry_setup(entry, component)
-        )
+        hass.async_create_task(hass.config_entries.async_forward_entry_setup(entry, component))
 
     @callback
     def async_node_added(node):
-        _LOGGER.debug("NODE ADDED: %s - node id %s", node, node.id)
+        _LOGGER.debug("[NODE ADDED] node_id: %s", node.id)
         data_nodes[node.id] = node
         data_values[node.id] = []
 
     @callback
     def async_node_changed(node):
-        _LOGGER.debug("NODE CHANGED: %s", node)
+        _LOGGER.debug("[NODE CHANGED] node_id: %s", node.id)
         data_nodes[node.id] = node
 
+    @callback
+    def async_node_removed(node):
+        _LOGGER.debug("[NODE REMOVED] node_id: %s", node.id)
+        # Remove entities belonging to this node
+        node_data_values = data_values[node.id]
+        for values in node_data_values:
+            if values.entity:
```

Comment:
We should avoid storing the entity instance on the values but just keep an id to the entity so we can signal it